cxx/pclean/schema_helper_test.cc

```diff
@@ -218,6 +247,83 @@ BOOST_AUTO_TEST_CASE(test_make_hirm_schmea) {
   // "City" moved to the front of the list.
   expected_domains = {"City", "School", "Physician", "Practice", "Record"};
   BOOST_TEST(nr5.domains == expected_domains, tt::per_element());
+
+  BOOST_TEST(tschema.contains("Physician:school::School:name"));
+  BOOST_TEST(tschema.contains("Practice:city::City:name"));
```

Comment:
What if we have something like 

```
class Practice
  location ~ City
  previous_location ~ City
```
? That seems like something we should support.

Reply:
We do support it!  Assuming these are both observed in the query section, they will generate intermediate noisy relations named "Practice:location::City:name" and "Practice:previous_location::City:name".